OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -197,8 +201,24 @@ void DeployMcv(IBot bot, Actor mcv, bool move)
 
 			var baseCenter = GetRandomBaseCenter();
 
-			return FindPos(baseCenter, baseCenter, Info.MinBaseRadius,
+			// Select a Spot in the base.
+			var selectedPosition = FindPos(baseCenter, baseCenter, Info.MinBaseRadius,
 				distanceToBaseIsImportant ? Info.MaxBaseRadius : world.Map.Grid.MaximumTileSearchRange);
+
+			// Otherwise select a location near resource fields.
+			if (resourceLayer != null && !distanceToBaseIsImportant)
+			{
+				var nearbyResources = world.Map.FindTilesInAnnulus(selectedPosition.Value,
+					world.Map.Grid.MaximumTileSearchRange - 2, world.Map.Grid.MaximumTileSearchRange)
+					.Where(a => resourceLayer.GetResource(a).Type != null)
+					.Shuffle(world.LocalRandom).Take(6);
```

Comment:
where's the handling for the case where no resources are found? Won't this crash?